blueocean-pipeline-editor/src/main/js/EditorPage.jsx

```diff
@@ -323,6 +323,18 @@ class PipelineLoader extends React.Component {
         });
     }
 
+    checkForNestedSequentialStages(internalModelPipeline, handler) {
+        for (let i = 0; i < internalModelPipeline.children.length; i++) {
+            for (let j = 0; j < internalModelPipeline.children[i].children.length; j++) {
+                if (typeof internalModelPipeline.children[i].children[j].stages !== 'undefined') {
```

Comment:
Should we also have `internalModelPipeline.children[i].children[j].stages.length > 0` as check?

Reply:
In the pipeline example @abayer wrote, after converting the pipeline to internalModel, internalModelPipeline.children[i].children[j].stages actually has a type of obj, not array, so it doesn't have length, it does have an array called json, but because I am not sure that will always be present, I decided against testing for it.

From my testing, internalModelPipeline.children[i].children[j].stages is always undefined for all my test cases except sequential parallel stages

![internal-model-structure](https://user-images.githubusercontent.com/2933351/41130341-9b18ea40-6af9-11e8-8572-b41e6623400d.png)